README.md

```diff
@@ -1,2 +1,98 @@
-# http-message-signing-proxy
-An HTTP proxy intended for use with the Form3 API that signs incoming requests using HTTP Message Signatures RFC
+# HTTPS signing proxy
+
+- [Introduction](#introduction)
+- [Run the proxy](#run-the-proxy)
+- [Configuration](#configuration)
+  - [Configuration override](#configuration-override)
+    - [Using flags](#override-config-using---set-flag)
+    - [Using env var](#override-config-using-env-var)
+- [Proxy mechanism](#proxy-mechanism)
+- [Metrics](#metrics)
+
+## Introduction
+
+The request signing proxy would sit in front of the client and intercept outbound requests, sign them with client's
+private key and transfer signed requests to the server. It uses the request signing library [here](https://github.com/form3tech-oss/go-http-message-signatures).
+
+![design.png](doc/images/design.png)
+
+More about request signing [here](https://api-docs.form3.tech/tutorial-request-signing.html).
+
+## Run the proxy
+
+The proxy requires `--config` flag, which point to the config file. See [configuration](#configuration) section below.
+
+```shell
+./signing-proxy --config <config_file_path>
+```
+
+In [example directory](./example), there are several files which help running the proxy locally:
+
+- [config_example.yaml](./example/config_example.yaml): A complete config file that should be in `--config` flag.
+- [cert.crt](./example/cert.crt): A self-signed certificate for an HTTPS proxy (TLS mode only).
+- [private.key](./example/private.key): The private key associated with the certificate above (TLS mode only).
+- [rsa_private_key.pem](./example/rsa_private_key.pem): A private key that is used to sign incoming request.
+- [rsa_public_key.pub](./example/rsa_public_key.pub): The public key associated with the private key above. It is not 
+  used by the proxy but will be useful if you need to verify the signature.
+- [docker-compose.yaml](./example/docker-compose.yaml): A convenient docker compose file that runs the proxy with all 
+  the config above. The proxy serves HTTPS requests at port 8080.
```

Comment:
We could mention the example `Dockerfile` here as a shortcut, so people can just jump to the file quickly

Reply:
added above